docs/DeveloperGuide.md

```diff
@@ -296,6 +296,151 @@ The following sequence diagram shows how the flow of the example execution:
+
+Given this flexibility and variety of valid user input, it is a challenge to parse all form of valid user input correctly and efficiently. 
+
+The implementation of the `EditPersonCommand` and `EditPersonCommandParser` can serve as a good example to represent how we tackle this challenge at command and parser class level to achieve the desired flexibility and usability while keeping the code clean and maintainable.
+
+#### Implementation
+The parsing and execution process of the `EditPersonCommand` is achieved via the combination of Four groups of classes: `AddressBookParser`, `TypeParsingUtil`, `EditPersonCommandParser` and `EditPersonCommand`, each of which is responsible for a different stage of the parsing process. 
+
+Given a user input that is intended to be parsed into an `EditPersonCommand`, a high level description of the parsing process is as follows:
+
+The first stage is to understand that user intends to invoke the `EditPersonCommand` and delegate the parsing to the specialised parser class, `EditPersonCommmandParser`. This step is done by the `AddressBookParser` class, which behave like a "simple factory" via a giant "switch" statement. We will not discuss it in depth here as it is not very interesting.
+
+The second stage is to parse each Parameter and flags. This work is delegated to the `typeParsingUtil` class, which is a utility class that contains many static methods that are responsible for parsing different types of user input that is reused across different command parsers.
+
+Then, the `EditPersonCommandParser` class is responsible for combining the results of the previous stage and construct the `EditPersonCommand` object.
+
+Finally, the `EditPersonCommand` class is responsible for executing the command and updating the model accordingly.
+
+The following section will discuss the implementation of the later three groups of classes in more detail.
+
+##### 1. ListEntry and ListEntryField class
+In TutorMates, Lesson and Person extends `ListEntry` class as they are displayed as items of the ___STUDENTS list___, ___SCHEDULE list___ respectively.
+
+Each `ListEntry` object contains a list of `ListEntryField` objects, which are the fields of the `ListEntry` object. For example, a `Person` object contains a list of `ListEntryField` objects, which are the fields of the `Person` object, such as name, phone number, email address, etc.
+
+These two classes are used in multiple generic class and methods to achieve the flexibility and robustness of the parsing process.
+
+##### 2. Parsing of Flags and Parameters
+The `TypeParsingUtil` class parse all kinds of flags (which are all `ListEntryField` ) via a single powerful generic method, `parseField`. The method signature is as follows:
```

Comment:
parse -> parses

Reply:
Thank you for pointing that out. Fixed in the latest commit.